lib/ocpp/common/websocket/websocket_libwebsockets.cpp

```diff
@@ -1288,4 +1294,31 @@ int WebsocketTlsTPM::process_callback(void* wsi_ptr, int callback_reason, void*
     return 0;
 }
 
+void WebsocketTlsTPM::push_deferred_callback(const std::function<void()>& callback) {
+    std::scoped_lock tmp_lock(this->deferred_callback_mutex);
+    this->deferred_callback_queue.push(callback);
+    this->deferred_callback_cv.notify_one();
+}
+
+void WebsocketTlsTPM::handle_deferred_callback_queue() {
+    while (true) {
+        std::function<void()> callback;
+        {
+            std::unique_lock lock(this->deferred_callback_mutex);
+            this->deferred_callback_cv.wait(
```

Comment:
While the condition is correct, please use a **wait_for** (like the recv_loop) with a timeout of 1 second. It should be enough to free up CPU and still sleep most of the time when we don't have a callback in the queue.

Reply:
I've changed this but I still don't like it since now the following code also needs to be changed since I can get past the `wait_for` even if there is nothing to do. I am really curious about your experience with the 0.01% of the cases where the notification gets lost. I did not find any similar experiences, the only thing is that people can implement the variable incorrectly or use the wrong mutex and stuff like that.

I'd rather leave it a `wait` instead since that is what we are always doing here.

Reply:
Any thoughts here @AssemblyJohn

Reply:
Except the previous discussion I have no further comments.